File: src/main/java/org/timmc/johnny/UrlDecodeException.java
package org.timmc.johnny;

/**
 * Runtime exception denoting a failure to recognize or parse a URL or some part of a URL.
 */
public class UrlDecodeException extends RuntimeException {
    public UrlDecodeException(String msg) { super(msg); }
    public UrlDecodeException(String msg, Throwable cause) { super(msg, cause); }
    public UrlDecodeException(Throwable cause) { super(cause); }
}

File: src/main/java/org/timmc/johnny/UrlDecodeException.kt
package org.timmc.johnny

/**
 * Runtime exception denoting a failure to recognize or parse a URL or some part of a URL.
 */
class UrlDecodeException: RuntimeException {
    constructor(msg: String) : super(msg)
    constructor(msg: String, cause: Throwable) : super(msg, cause)
}
